sql/plan/innerjoin.go

```diff
@@ -127,3 +125,74 @@ func (j *InnerJoin) TransformExpressions(f sql.TransformExprFunc) (sql.Node, err
 
 	return NewInnerJoin(j.Left, j.Right, cond), nil
 }
+
+type innerJoinIter struct {
+	l    sql.RowIter
+	rp   rowIterProvider
+	r    sql.RowIter
+	ctx  *sql.Context
+	cond sql.Expression
+
+	leftRow sql.Row
+}
+
+func (i *innerJoinIter) Next() (sql.Row, error) {
+	for {
+		if i.leftRow == nil {
+			r, err := i.l.Next()
+			if err != nil {
+				return nil, err
+			}
+
+			i.leftRow = r
+		}
+
+		if i.r == nil {
+			iter, err := i.rp.RowIter(i.ctx)
+			if err != nil {
+				return nil, err
+			}
+
+			i.r = iter
+		}
+
+		rightRow, err := i.r.Next()
+		if err == io.EOF {
+			i.r = nil
+			i.leftRow = nil
+			continue
+		}
+
+		if err != nil {
+			return nil, err
+		}
+
+		var row sql.Row
```

Comment:
actually it looks like we can just write:
```
i.leftRow = append(i.leftRow, rightRow...)
```
am I wrong?

Reply:
and eval at the end

Reply:
We cannot append rightRow to leftRow.

When we iterate we iterate the left side only once and for each left row, we get many right rows.

So the result row is Left row + Right row.

If we do 
```i.leftRow = append(i.leftRow, rightRow...)```

Left row would change each iteration, which is not what we want.